### PR TITLE
HPCC-22889 Add Query Associated Files to ZAP report

### PR DESCRIPTION
**Pull request by wangkx**
Query Associated Files may be used to import WU from
ZAP report.

Signed-off-by: wangkx <kevin.wang@lexisnexis.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [x] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [x] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [ ] This is a user interface / front-end modification
  - [ ] I have tested my changes in multiple modern browsers
  - [ ] The component(s) render as expected

## Smoketest:
- [ ] Send notifications about my Pull Request position in Smoketest queue.
- [ ] Test my draft Pull Request.

### CODE REVIEW

**Comment by wangkx** [comment]
@RussWhitehead please review.

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp [hunk 1]
```diff
@@ -3721,6 +3722,51 @@ void CWsWuFileHelper::createWUZAPFile(IEspContext& context, IConstWorkUnit* cwu,
...
+        {
+            IERRLOG("Cannot open %s on %s.", name.str(), ip.str());
+            continue;
+        }
+
+        StringBuffer fileName(name.str());
+        getFileNameOnly(fileName, false);
+
+        StringBuffer outFileName(pathToCreate);
+        outFileName.append(PATHSEPCHAR).append(fileName);
```
Comment (RussWhitehead) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp]: What happens here is pathToCreate is PATHSEPCHAR ("\")?  Will you end up with "\\" ?

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp [hunk 2]
```diff
@@ -3721,6 +3722,51 @@ void CWsWuFileHelper::createWUZAPFile(IEspContext& context, IConstWorkUnit* cwu,
         throw MakeStringException(ECLWATCH_CANNOT_COMPRESS_DATA,"Failed to execute system command 'zip'. Please make sure that zip utility is installed.");
 }
 
+void CWsWuFileHelper::createZAPWUQueryAssociatedFiles(IConstWorkUnit* cwu, const char* pathToCreate)
+{
+    Owned<IConstWUQuery> query = cwu->getQuery();
```
Comment (RussWhitehead) [inline esp/services/ws_workunits/ws_workunitsHelpers.cpp]: Any need to check for cwu == nullptr or pathToCreate == nullptr ?

**Review by RussWhitehead — COMMENTED** [review]
A couple minor questions

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp (hunk 1 above)
Comment (wangkx) [reply esp/services/ws_workunits/ws_workunitsHelpers.cpp]: The pathToCreate is set in the createWorkingFolder(). It is 'tempzipfiles/ZAPReport_wuid_user'.

File: esp/services/ws_workunits/ws_workunitsHelpers.cpp (hunk 2 above)
Comment (wangkx) [reply esp/services/ws_workunits/ws_workunitsHelpers.cpp]: The cwu has been checked at the begin of creating ZAP report (CWsWorkunitsSoapBindingEx:: createAndDownloadWUZAPFile). The pathToCreate cannot be nullptr (see my previous comment).

**Comment by RussWhitehead** [comment]
@wangkx Looks good

**Comment by wangkx** [comment]
@richardkchapman this PR is read to be merged.

**Pull request merged**